extensions/src/main/java/io/lionweb/repoclient/RepoSerialization.java

```diff
@@ -0,0 +1,261 @@
+package io.lionweb.repoclient;
+
+import static io.lionweb.lioncore.java.serialization.LowLevelJsonSerialization.groupNodesIntoSerializationBlock;
+
+import io.lionweb.lioncore.java.serialization.LowLevelJsonSerialization;
+import io.lionweb.lioncore.java.serialization.data.SerializedChunk;
+import io.lionweb.lioncore.java.serialization.data.SerializedClassifierInstance;
+import io.lionweb.repoclient.api.BulkAPIClient;
+import io.lionweb.repoclient.api.RawBulkAPIClient;
+import io.lionweb.serialization.extensions.AdditionalAPIClient;
+import io.lionweb.serialization.extensions.BulkImport;
+import io.lionweb.serialization.extensions.Compression;
+import io.lionweb.serialization.extensions.TransferFormat;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+import kotlin.text.Charsets;
+
+/** This class contains the logic to store and retrieve entire repositories at once. */
+public class RepoSerialization {
+  private int nNodesThreshold = 100_000;
```

Comment:
typo? "nNodes"

Reply:
It stands for number of nodes. I have renamed it to `numberOfNodesThreshold`